base/dsfs/dataset.go

```diff
@@ -470,16 +509,49 @@ func setChecksumAndLength(ds *dataset.Dataset, data qfs.File, buf *bytes.Buffer,
 }
 
 // returns a commit message based on the diff of the two datasets
-func generateCommitDescriptions(prev, ds *dataset.Dataset, force bool) (short, long string, err error) {
+func generateCommitDescriptions(store cafs.Filestore, prev, ds *dataset.Dataset, bodyAct BodyAction, forceIfNoChanges bool) (short, long string, err error) {
 	if prev == nil || prev.IsEmpty() {
 		return "created dataset", "created dataset", nil
 	}
 
 	ctx := context.TODO()
 
-	// TODO(dlong): Inline body if it is a reasonable size, in order to get information about
-	// how the body has changed.
-	// TODO(dlong): Also should ignore derived fields, like structure.{checksum,entries,length}.
+	// Inline body if it is a reasonable size, to get message about how the body has changed.
+	if bodyAct != BodySame {
+		// If previous version had bodyfile, read it and assign it
+		if prev.Structure != nil && prev.Structure.Length < BodySizeSmallEnoughToDiff {
+			if prev.BodyFile() != nil {
+				prevReader, err := dsio.NewEntryReader(prev.Structure, prev.BodyFile())
+				if err == nil {
+					prevBodyData, err := readAllEntries(prevReader)
+					if err == nil {
+						prev.Body = prevBodyData
+					}
+				}
+			}
+		} else {
+			bodyAct = BodyTooBig
+		}
+	}
+
+	// Read the transform files to see if they changed.
+	// TODO(dustmop): Would be better to get a line-by-line diff
+	if prev.Transform != nil && prev.Transform.ScriptPath != "" {
+		err := prev.Transform.OpenScriptFile(ctx, store)
+		if err == nil {
+			tfFile := prev.Transform.ScriptFile()
+			prev.Transform.ScriptBytes, _ = ioutil.ReadAll(tfFile)
+		}
+	}
+	if ds.Transform != nil && ds.Transform.ScriptPath != "" {
+		// TODO(dustmop): The ipfs filestore won't recognize local filepaths, we need to use
+		// local here. Is there some way to have a cafs store that works with both?
```

Comment:
yes. Thats covered in the interface that's intended to replace `cafs.Filestore`. 

`qfs.Filesystem` has a `Mux` implementation for exactly this: https://github.com/qri-io/qfs/blob/a6a7b98f0fa0969f6eae50fe14c2c6b5b32f4b4f/mux.go#L13-L17

To get there we'd need to do a _lot_ of refactoring.